dali/kernels/imgproc/warp/warp_setup.cuh

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_WARP_WARP_SETUP_CUH_
+#define DALI_KERNELS_IMGPROC_WARP_WARP_SETUP_CUH_
+
+#include <vector>
+#include <utility>
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/common/block_setup.h"
+#include "dali/core/geom/vec.h"
+
+namespace dali {
+namespace kernels {
+namespace warp {
+
+template <int ndim, typename OutputType, typename InputType>
+struct SampleDesc {
+  OutputType *__restrict__ output;
+  const InputType *__restrict__ input;
+  ivec<ndim> out_size, out_strides, in_size, in_strides;
+  int channels;
+  DALIInterpType interp;
+};
+
+
+template <int ndim, typename OutputType, typename InputType>
+class WarpSetup : public BlockSetup<ndim, ndim> {
```

Comment:
So, I have some questions about this WarpSetup thing.
1. What is it supposed to do?
2. What is the benefit of using inheritance here? It doesn't seem to be neccessary, as all the functions can be called keeping BlockSetup in composition-like manner, plus I couldn't find any reason that either WarpSetup or WarpGPU should be treated as a BlockSetup.

Reply:
Ad 1. It provides mapping-agnostic setup for warp operations.
Ad 2: It's more complicated. I was even considering using the name `WarpBase` for what is now `WarpSetup`. I guess you wouldn't complain then ;)
Why WarpSetup inherits BlockSetup - because from the kernel perspective the blocks, samples and grid are all parts of the setup - the fact that block setup is abstracted away is of no concern here.
It's all about inheriting implementation - and I would really prefer to have private inheritance (Warp from WarpSetup), but currently I need public access to the setup in tests.

Reply:
Made it a member.